### PR TITLE
Add sort options for citations

### PR DESCRIPTION
**Pull request by bluebear94**
Mentioned in some comments to #1713.

### CODE REVIEW

File: crates/typst-library/src/meta/bibliography.rs [hunk 1]
```diff
@@ -565,16 +590,10 @@ fn create(bibliography: BibliographyElem, citations: Vec<CiteElem>) -> Arc<Works
...
-        BibliographyStyle::Apa => Box::new(style::Apa::new()),
-        BibliographyStyle::ChicagoAuthorDate => Box::new(style::ChicagoAuthorDate::new()),
-        BibliographyStyle::ChicagoNotes => Box::new(style::ChicagoNotes::new()),
-        BibliographyStyle::Ieee => Box::new(style::Ieee::new()),
-        BibliographyStyle::Mla => Box::new(style::Mla::new()),
-    };
-
-    let references = db
-        .bibliography(&*bibliography_style, None)
+    // Recompute this, as prior Database::citation calls might have
```
Comment (laurmaedje) [inline crates/typst-library/src/meta/bibliography.rs:593]: If we recompute this here, are we sure that the order is still the same.

**Comment by bluebear94** [comment]
Actually, I’d realized that `Database::citation` can take multiple citation entries (see [my comment in #1713](https://github.com/typst/typst/issues/1713#issuecomment-1635322991)). I wonder whether it would be more appropriate to take advantage of this instead of passing each citation entry in the element, one at a time.

**Comment by laurmaedje** [comment]
I'm trying to remember why I didn't make use of this feature in the first place. I think it had something to with the brackets or maybe supplements, but I'm not sure.

**Comment by laurmaedje** [comment]
Do you think the alternative way to use `Database::citation` blocks this PR?

**Comment by bluebear94** [comment]
Yes, I’ll open another one that does that.

**Comment by bluebear94** [comment]
> I'm trying to remember why I didn't make use of this feature in the first place. I think it had something to with the brackets or maybe supplements, but I'm not sure.

It seems to have to do with not only brackets but also links (each reference in the cite elem should link to the respective bibliography entry).

**Comment by laurmaedje** [comment]
I'm going to close this for now. Hayagriva will likely get a larger update when @reknih integrates CSL support soon and we can see after if/how this is still relevant. Thanks!

**Pull request closed without merging**